### ISSUE TITLE
refactor: remove unused wait=False path from build_systems

### ISSUE THREAD

**Issue by gregorweiss** [emdgroup/mdfactory#52]
## Problem

`build.py` has the same `wait=False` / `session.detach()` pattern that was just removed from `simulate.py` in PR 50:

```python
if not wait:
    logger.warning("Returning raw futures — caller must call parsl.clear() when done.")
    session.detach()
    return futures
```

Same issues apply:
- **No callers use it** — every call site passes `wait=True` (the default)
- **Broken return type** — returns raw `AppFuture` objects that callers can't meaningfully use without managing the Parsl session themselves
- **Leaks session** — `session.detach()` leaves the Parsl session open with no cleanup guarantee

## Proposed fix

- Remove the `wait` parameter from `build_systems`
- Remove the 4-line `if not wait` block and `session.detach()` call
- Remove the `wait` parameter from the docstring
- Update any tests that use `wait=False`

~15 lines removed, no behavioral change for any existing caller.

## Acceptance criteria

- [ ] `wait` parameter removed from `build_systems` signature and docstring
- [ ] `if not wait` block and `session.detach()` call removed
- [ ] All existing tests pass
- [ ] No behavioral changes
